pkg/kubectl/service_basic.go

```diff
@@ -189,9 +189,6 @@ func (s ServiceCommonGeneratorV1) validate() error {
 	if s.ClusterIP == api.ClusterIPNone && s.Type != api.ServiceTypeClusterIP {
 		return fmt.Errorf("ClusterIP=None can only be used with ClusterIP service type")
 	}
-	if s.ClusterIP == api.ClusterIPNone && len(s.TCP) > 0 {
```

Comment:
API validation doesn't validate that the port and targetPort should be equal for headless service. The code is commented currently.

https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L2866-L2873
```
	// in the v1 API, targetPorts on headless services were tolerated.
	// once we have version-specific validation, we can reject this on newer API versions, but until then, we have to tolerate it for compatibility.
	//
	// if isHeadlessService {
	// 	if sp.TargetPort.Type == intstr.String || (sp.TargetPort.Type == intstr.Int && sp.Port != sp.TargetPort.IntValue()) {
	// 		allErrs = append(allErrs, field.Invalid(fldPath.Child("targetPort"), sp.TargetPort, "must be equal to the value of 'port' when clusterIP = None"))
	// 	}
	// }
```